src/IdentityServer8/src/Services/Default/DefaultKeyMaterialService.cs

```diff
@@ -39,7 +39,7 @@ public async Task<SigningCredentials> GetSigningCredentialsAsync(IEnumerable<str
     {
         if (_signingCredentialStores.Any())
         {
-            if (allowedAlgorithms.IsNullOrEmpty())
+            if (allowedAlgorithms != null && !allowedAlgorithms.Any())
```

Comment:
Should use the extension method EnumerableIsNullOrEmpty. In IdentityServer4 it called IsNullOrEmpty, but this causes a conflict in .NET 8, so the extension method was renamed EnumerableIsNullOrEmpty in IdentityServer8.